src/coreComponents/linearAlgebra/DofManager.cpp

```diff
@@ -364,10 +487,28 @@ void DofManager::addCoupling( string const & fieldName,
 
   CouplingDescription & coupling = m_coupling[ {fieldIndex, fieldIndex} ];
   coupling.connector = Connector::Stencil;
-  coupling.regions = field.regions;
+  coupling.support = field.support;
   coupling.stencils = &stencils;
 }
 
+void DofManager::addCoupling( string const & rowFieldName,
+                              string const & colFieldName,
+                              DofManager::Connector connectivity,
+                              map< string, array1d< string > > const & regions,
+                              bool symmetric )
+{
+  // Convert input into internal format
+  std::vector< Regions > support;
+  for( auto const & p : regions )
+  {
+    MeshBody const & meshBody = m_domain->getMeshBody( p.first );
+    MeshLevel const & mesh = meshBody.getMeshLevel( 0 ); // TODO by name?
+    std::vector< string > regionNames( p.second.begin(), p.second.end() );
+    support.push_back( { meshBody.getName(), mesh.getName(), std::move( regionNames ) } );
```

Comment:
@klevzoff ?? I don't know that anything bad can happen by using the initialization list? I will concede the constructor is more obvious, but I don't think either way would save us if parameters are swapped.

Reply:
Regardless of whether `Regions` has a constructor or not, the two lines do exactly the same thing (call the constructor if one exists, or aggregate-initialize if not). There's also not a way (that I know of) to prevent the user from accidentally mixing up two consecutive parameters of the same type, short of using a separate "strong type" for each parameter, i.e. 
```c++
struct MeshBodyName : public std::string { using std::string::string; };
```
(but note that this can also be done with an aggregate struct without constructors).

The case @TotoGaz mentions where someone changes the layout of `Regions` could indeed be helped by having a constructor, provided that the same person doesn't simultaneously change the order of constructor arguments. However, purely on philosophical grounds, I believe a struct with all public members that does not maintain any invariants (in other words, serves simply to group related data, like a `tuple`) _should not_ have constructors. If we decided `Regions` should have invariants or other useful logic (say, check that given names correspond to an existing mesh body/level), I'd fully agree to adding a constructor, but here it's really used as a more expressive and compile-time-friendly replacement for `tuple`.

Reply:
_In that context_, I believe that the difference between a `tuple` and a `struct` is that
- the first relies on explicit _indexing_ to get the proper information
- while the latter relies on explicit "_keys_" (the attribute names).

That's fair and the contracts are clear.
But by using the "_aggregate-initialize_" for `Regions` we now rely on both (adding _implicit_ indexing).
So we have to be careful for both.

_Nobody_ will ever think and see that reordering the parameters (some renaming + sorting in alphabetical order, _e.g._) will break the code in another file.
But _everybody_ knows that changing the signature of a function needs to be  propagated. Tools provided by the IDEs will even do this for you.

To face this issue, I've just discovered that c++20 introduces `designated initializers` (see [here](https://en.cppreference.com/w/cpp/language/aggregate_initialization)):
```c++
struct A { int x; int y; int z; };
A a{.y = 2, .x = 1}; // error; designator order does not match declaration order
A b{.x = 1, .z = 2}; // ok, b.y initialized to 0
```
Is acts like some on-the-fly constructor generation I presume.

It's a matter of risks. No approach will save us for sure. But one is really risky, the other is really safe.

Reply:
@TotoGaz c++20 it is then!!